tests/e2e/tkgs_ha.go

```diff
@@ -488,4 +344,785 @@ var _ = ginkgo.Describe("[csi-tkgs-ha] Tkgs-HA-SanityTests",
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				err = fpv.DeletePersistentVolume(client, staticPv.Name)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				ginkgo.By("Verify PVs, volumes are deleted from CNS")
+				err = fpv.WaitForPersistentVolumeDeleted(client, staticPv.Name, framework.Poll,
+					framework.PodDeleteTimeout)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeID)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+					fmt.Sprintf("Volume: %s should not be present in the CNS after it is deleted from "+
+						"kubernetes", volumeID))
+				ginkgo.By("Verify volume is deleted in Supervisor Cluster")
+				volumeExists := verifyVolumeExistInSupervisorCluster(svcPv[0].Spec.CSI.VolumeHandle)
+				gomega.Expect(volumeExists).To(gomega.BeFalse())
+
+			}()
+
+			ginkgo.By("Verify SV storageclass points to GC storageclass")
+			gomega.Expect(*svPvclaim.Spec.StorageClassName == storageclass.Name).To(
+				gomega.BeTrue(), "SV storageclass does not match with gc storageclass")
+			framework.Logf("GC PVC's storageclass matches SVC PVC's storageclass")
+
+			ginkgo.By("Verify GV PV has has required PV node affinity details")
+			_, err = verifyVolumeTopologyForLevel5(staticPv, allowedTopologyHAMap)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("GC PV: %s has required Pv node affinity details", staticPv.Name)
+
+			ginkgo.By("Verify SV PV has has required PV node affinity details")
+			_, err = verifyVolumeTopologyForLevel5(svcPv[0], allowedTopologyHAMap)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("SVC PV: %s has required PV node affinity details", svcPv[0].Name)
+
+			ginkgo.By("Create a pod and verify pod gets scheduled on appropriate " +
+				"nodes preset in the availability zone")
+			pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{staticPvc}, false, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			defer func() {
+				ginkgo.By("Delete pod")
+				err = fpod.DeletePodWithWait(client, pod)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}()
+
+			_, err = verifyPodLocationLevel5(pod, nodeList, allowedTopologyHAMap)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		})
+
 	})
+
+// verifyVolumeProvisioningWithServiceDown brings the service down and creates the statefulset and then brings up
+// the service and validates the volumes are bound and required annotations and node affinity are present
+func verifyVolumeProvisioningWithServiceDown(serviceName string, namespace string, client clientset.Interface,
+	storagePolicyName string, allowedTopologyHAMap map[string][]string, categories []string, isServiceStopped bool,
+	f *framework.Framework) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ginkgo.By("CNS_TEST: Running for GC setup")
+	nodeList, err := fnodes.GetReadySchedulableNodes(client)
```

Comment:
You are already checking this in Before each. You can remove it here

Reply:
Done